### PR TITLE
Statically linked the MSVC CRT

### PR DESCRIPTION
**Pull request by alexcrichton**
This PR updates a few dependencies for support with `-C
target-feature=+crt-static` and then updates CI to produce those relevant
binaries on Windows. This should produce binaries that don't require the MSVC
redistributables and instead work standalone.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 environment:
+  RUSTFLAGS: -Zunstable-options -Ctarget-feature=+crt-static
   matrix:
   - TARGET: i686-pc-windows-msvc
     BUILD_MSI: 1
   - TARGET: i686-pc-windows-gnu
-    MINGW_URL: https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win32/Personal%20Builds/mingw-builds/4.9.2/threads-win32/dwarf/i686-4.9.2-release-win32-dwarf-rt_v4-rev4.7z/download
+    MINGW_URL: https://s3.amazonaws.com/rust-lang-ci
```
Comment (alexcrichton) [inline appveyor.yml:7]: This was a bit of an unrelated cleanup, I switched to the url that we're using for Rust and a few other projects, and AWS is generally much more reliable than sourceforge

**Comment by brson** [comment]
This has some conflicts with the openssl-sys dep used by rustup's download crate.

**Comment by brson** [comment]
I may spend some time cleaning up the download crate soon to unblock this.

**Comment by alexcrichton** [comment]
Ok, updated the PR

**Comment by brson** [comment]
openssl-sys still not working right

**Comment by alexcrichton** [comment]
Pushed some updates, let's see how it goes
